### PR TITLE
Consistent use of configuration

### PR DESCRIPTION
**Pull request by difince**
`/res/configuration.toml` file added
prerequisite - existence of /$HOME/.edgex-cli/configuration.toml file !!!! (manually copy `res/configuration.toml` file into  `/$HOME/.edgex-cli/` directory) (this should be fixed)
remove hardcoded values of` Ports`, `Hosts`
use go-mod-core-contracts -> clients, thus removing hardcoded  `Paths` values 
"goimport -w ." executed to order the imports
refactoring

Fix: https://github.com/edgexfoundry-holding/edgex-cli/issues/135, https://github.com/edgexfoundry-holding/edgex-cli/issues/141

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

### CODE REVIEW

File: config/configuration.go [hunk 1]
```diff
@@ -16,162 +16,50 @@ package config
...
+	"strings"
 
-	"github.com/spf13/afero"
-	"gopkg.in/yaml.v2"
+	"github.com/BurntSushi/toml"
+	"github.com/spf13/viper"
 )
 
-var AppFs = afero.NewOsFs()
+var confDir = ".edgex-cli"
```
Comment (mkbhanda) [inline config/configuration.go:27]: If we assume for the moment that we are running out of a git repo, then .edge-cli would be in our git directory. That simplifies a lot of things. The whole home directory hidden file is a complication to support EdgeX-cli in our command path and then needing a fixed location for the config file but that becomes moot when we allow specifying a config file.

File: pkg/client.go [hunk 2]
```diff
@@ -11,21 +11,20 @@ import (
 
 var client = &http.Client{}
 
-func buildURL(itemType string, path string, port string) string {
+// deprecated
+func buildURL(itemType string, path string, port int) string {
 	host := viper.GetString("host")
```
Comment (mkbhanda) [inline pkg/client.go:16]: where/how does this host get set?

File: cmd/db/purge/purge.go [hunk 3]
```diff
@@ -115,31 +119,24 @@ func purge() {
...
+	ctx, _ := context.WithCancel(context.Background())
 
-	deviceData, err := client.GetAllItems("device", "48081")
+	url := config.Conf.Clients["Metadata"].Url() + clients.ApiDeviceRoute
+	mdc := metadata.NewDeviceClient(local.New(url))
 
+	devices, err := mdc.Devices(ctx)
 	if err != nil {
-		fmt.Println(err)
+		fmt.Errorf(err.Error())
```
[the remark this answers is not in the document]
Comment (difince) [reply cmd/db/purge/purge.go:129]: It looks so. Ok , anyway we need to figure out the most appropriate way to handle and print the error. I will now revert it

File: cmd/interval/list/list.go [hunk 4]
```diff
@@ -57,19 +55,18 @@ func listHandler(cmd *cobra.Command, args []string) {
 
 	data, _ := ioutil.ReadAll(resp.Body)
 
-	intervalList := intervalList{}
-	var errjson error
+	var intervals []models.Interval
 	var interval models.Interval
 	if len(args) > 0 {
-		errjson = json.Unmarshal(data, &interval)
+		err = json.Unmarshal(data, &interval)
```
[the remark this answers is not in the document]
Comment (difince) [reply cmd/interval/list/list.go:61]: it looks only specific interval could be printed vs all intervals
But yes, it does not looks good . 
Could address it in another PR

File: cmd/status/status.go [hunk 5]
```diff
@@ -37,38 +39,20 @@ This command pings each edgex microservice and prints their status.
...
-			microservices["Rules Engine"] = "48075"
-			microservices["Client Registration"] = "48071"
-			microservices["System Management"] = "48090"
-
 			w := new(tabwriter.Writer)
 			w.Init(os.Stdout, 0, 8, 1, '\t', 0)
 
-			for microservice, port := range microservices {
-				resp, err := http.Get("http://" + host + ":" + port + "/api/v1/ping")
+			for clientName, client := range config.Conf.Clients {
```
[the remark this answers is not in the document]
Comment (difince) [reply cmd/status/status.go:45]: just for consistency with the rest of edgex .. no other reason

**Comment by difince** [comment]
> Puzzled if the global flags in root.go are ever used.
Yes, they are use, but not consistently in the entire project. This need to be fix

**Comment by mkbhanda** [comment]
Merged it because it improves on the situation significantly.

**Pull request merged**